src/dbsync/src/dbsyncPipelineFactory.cpp

```diff
@@ -0,0 +1,196 @@
+                dispatchResult(result);
+            }
+        }
+        void getDeleted(ResultCallback /*callback*/) override
+        {
+            if (m_spSyncNode)
+            {
+                m_spSyncNode->rundown();
+            }
+            // DBSyncImplementation::instance().getDeleted(m_handle, m_txnContext, calback);
+        }
+    private:
+        using SyncResult = std::tuple<ReturnTypeCallback, nlohmann::json>;
+        using DispatchCallbackNode = Utils::ReadNode<SyncResult>;
+        using SyncRowNode = Utils::ReadWriteNode<nlohmann::json, SyncResult, DispatchCallbackNode>;
+
+        std::shared_ptr<DispatchCallbackNode> getDispatchNode(const int threadNumber)
+        {
+            return std::make_shared<DispatchCallbackNode>
+            (
+                std::bind(&Pipeline::dispatchResult, this, std::placeholders::_1),
+                threadNumber ? threadNumber : std::thread::hardware_concurrency()
+            );
+        }
+        std::shared_ptr<SyncRowNode> getSyncNode(const int threadNumber)
+        {
+            return std::make_shared<SyncRowNode>
+            (
+                std::bind(&Pipeline::processSyncRow, this, std::placeholders::_1),
+                threadNumber ? threadNumber : std::thread::hardware_concurrency()
+            );
+        }
+
+        SyncResult processSyncRow(const nlohmann::json& value)
+        {
+            ReturnTypeCallback type{ DB_ERROR };
+            nlohmann::json result{ value };
+            try
+            {
+                // DBSyncImplementation::instance().syncTxRow(m_handle, m_txnContext, value, type, result);
+            }
+            catch(const DbSync::max_rows_error& /*ex*/)
+            {
+                result = MAX_ROWS;
+            }
+            catch(...)
+            {
+                //we'll notify with DB_ERROR.
+            }
+            return std::make_tuple<ReturnTypeCallback, std::string>(std::move(type), result[0]);
+        }
+        void dispatchResult(const SyncResult& result)
+        {
+            const auto& value{ std::get<1>(result) };
+            if (!value.empty())
+            {
+                m_callback(std::get<0>(result), value);
+            }
+        }
+        const std::shared_ptr<DispatchCallbackNode> m_spDispatchNode;
+        const std::shared_ptr<SyncRowNode> m_spSyncNode;
+        const DBSYNC_HANDLE m_handle;
+        const TXN_HANDLE m_txnContext;
+        const unsigned int m_maxQueueSize;
+        const ResultCallback m_callback;
+    };
+//----------------------------------------------------------------------------------------
+    PipelineFactory& PipelineFactory::instance()
+    {
+        static PipelineFactory s_instance;
+        return s_instance;
+    }
+    void PipelineFactory::release()
+    {
+        std::lock_guard<std::mutex> lock{ m_contextsMutex };
+        m_contexts.clear();
+    }
+    PipelineCtxHandle PipelineFactory::create(const DBSYNC_HANDLE handle,
```

Comment:
use noexcept for this function

Reply:
create can throw if dbsyncImplementation::createTransaction throws a json exception.